lib/gui/app/components/source-selector/source-selector.tsx

```diff
@@ -73,36 +77,158 @@ function getState() {
 	};
 }
 
-interface ImageSelectorProps {
+const URLSelector = ({ done }: { done: (imageURL: string) => void }) => {
+	const [imageURL, setImageURL] = React.useState('');
+	const [recentImages, setRecentImages]: [
+		string[],
+		(value: React.SetStateAction<string[]>) => void,
+	] = React.useState([]);
+	React.useEffect(() => {
+		const fetchRecentUrlImages = async () => {
+			try {
+				const recentUrlImages: any = await jsonStorage.get('recentUrlImages');
+				setRecentImages(recentUrlImages);
+			} catch (err) {
```

Comment:
Don't use a try catch here, this has to work.

Reply:
It relies on a method that has an error parameter in the callback and rightly so because it handles opening a file, and things might go wrong.
You reminded me though that we need a warning to pop-up instead of just logging the error, as the error won't probably trigger the dialogue by itself, given that it comes from a callback (so it has to be manually handled anyway)

Reply:
For now, when there's an error, nothing happens, the url prompt just stays there as if no button was clicked.